src/pgpx/connection.py

```
@@ -0,0 +1,190 @@
+    def connect(self) -> "DatabaseConnection":
+        """Establish database connection.
+
+        Returns:
+            Self for method chaining
+
+        Raises:
+            ConnectionError: If connection fails
+        """
+        try:
+            self._connection = psycopg.connect(**self.connection_params)
+            return self
```

Ruff (0.14.2)

Consider moving this statement to an `else` block

(TRY300)